### PR TITLE
Storage: add check for ceph crush map's mixed items buckets

### PR DESCRIPTION
**Pull request by taodd**
This patch will make hotsos able to report those crush buckets that contains mixed items.
mixed items bucket is always a problem which would cause ceph unable to recover

### CODE REVIEW

**Comment by taodd** [comment]
Ah, forgot to add one test file

**Comment by taodd** [comment]
Now done

File: plugins/storage/01ceph [hunk 1]
```diff
@@ -20,6 +20,49 @@ CEPH_SERVICES = [r"ceph-[A-Za-z]+",
 CEPH_INFO = {}
 
 
+def build_buckets_from_crushdump(crushdump):
```
Comment (dosaboy) [inline plugins/storage/01ceph:23]: Hi Dongdong,

I tried running this against a sosreport I have and got the following error:

Traceback (most recent call last):
  File "/home/ubuntu/git/dosaboy/hotsos/plugins/storage/01ceph", line 255, in <module>
    get_crushmap_mixed_buckets()
  File "/home/ubuntu/git/dosaboy/hotsos/plugins/storage/01ceph", line 45, in get_crushmap_mixed_buckets
    osd_crush_dump = helpers.get_osd_crush_dump()
  File "/home/ubuntu/git/dosaboy/hotsos/plugins/storage/common/helpers.py", line 30, in catch_exceptions_inner2
    return f(*args, **kwargs)
  File "/home/ubuntu/git/dosaboy/hotsos/plugins/storage/common/helpers.py", line 198, in get_osd_crush_dump
    return json.load(f)
  File "/usr/lib/python3.6/json/__init__.py", line 299, in load
    parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)
  File "/usr/lib/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.6/json/decoder.py", line 342, in decode
    raise JSONDecodeError("Extra data", s, end)
json.decoder.JSONDecodeError: Extra data: line 1 column 5 (char 4)

Looking at the contents of sos_commands/ceph/ceph_osd_crush_dump I see:

2021-04-05 15:19:45.664 7f4500a97700 -1 auth: unable to find a keyring on /etc/ceph/ceph.client.admin.keyring,/etc/ceph/ceph.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,: (2) No such file or directory
2021-04-05 15:19:45.664 7f4500a97700 -1 monclient: ERROR: missing keyring, cannot use cephx for authentication
[errno 2] error connecting to the cluster

This is pretty common if e.g. sosreport is from a ceph-osd node that does not have privileges/keys to get that info. So I think we need to protect against this when we decode the contents of the file. I'd lile to also see a unit tests that covers the case where sos_commands/ceph/ceph_osd_crush_dump exists but contains non-json data.

**Review by dosaboy — CHANGES_REQUESTED** [review]

File: plugins/storage/01ceph (hunk 1 above)
Comment (taodd) [reply plugins/storage/01ceph:23]: Hmm, right, needs to think about the un-expected non-json output

**Pull request merged**